atlasdb-api/src/main/java/com/palantir/atlasdb/persistent/api/PersistentStore.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.persistent.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+
+import org.immutables.value.Value;
+
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+
+public interface PersistentStore<K, V> extends AutoCloseable {
+    @Value.Immutable
+    interface EntryFamilyHandle {
+        UUID uniqueName();
+    }
+
+    /**
+     * Gets the value associated with the entry specified by {@code key}.
+     *
+     * @param entryFamilyHandle handle to the entry family from which we want to retrieve the value
+     * @param key                entry key for which we want to retrieve the value
+     * @return the {@link Optional} containing the value or empty if there is no associated value
+     * @throws SafeIllegalArgumentException when {@code entryFamilyHandle} is a handle to a non existing entry family
+     */
+    Optional<V> get(EntryFamilyHandle entryFamilyHandle, @Nonnull K key);
```

Comment:
Similar to how `Map` refers to `Map.Entry` within its class, you should refer to `Handle` as `PersistentStore.Handle` everywhere in this interface, and also in the implementations that have already had this.